### PR TITLE
San Francisco Working Families Tax Credit

### LINKED ISSUE #4578: San Francisco local EITC - Working Families Tax Credit
[$250 for filers eligible for the federal and state EITC
](https://www.sf.gov/news/san-francisco-launches-free-tax-preparation-assistance-and-local-tax-credit-program-and)

### PR DESCRIPTION
**Pull request by xianxinzhu**
Fixes #4578

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Add San Francisco working families tax credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - San Francisco working families tax credit.
```

File: policyengine_us/parameters/gov/states/ca/tax/income/credits/wftc/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+description: California provides a working family tax credit for filers who qualify for the federal or state EITC of this amount.
+values:
+  2023-01-01: 250
+metadata:
+  unit: currency-USD
+  period: year
+  label: California working family tax credit amount
+  reference:
+    - title: SF.Gov News 2024
+      href: https://www.sf.gov/news/san-francisco-launches-free-tax-preparation-assistance-and-local-tax-credit-program-and
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/credits/wftc/amount.yaml:10]: Can we try to find a tax form or legal code for all of the values

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/itemized/mt_medical_expense_deduction.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/itemized/mt_medical_expense_deduction.py:1]: This should not be in this file right?

File: policyengine_us/variables/gov/states/ca/tax/income/credits/wftc/ca_wftc.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from policyengine_us.model_api import *
+
+
+class ca_wftc(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/tax/income/credits/wftc/ca_wftc.py:4]:
```suggestion
class ca_sf_wftc(Variable):
```

File: policyengine_us/variables/gov/states/ca/tax/income/credits/wftc/ca_wftc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/tax/income/credits/wftc/ca_wftc.py:1]: move to local folder - all parameters / variables and tests

File: policyengine_us/variables/gov/states/ca/tax/income/credits/earned_income/ca_is_qualifying_child_for_caleitc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/tax/income/credits/earned_income/ca_is_qualifying_child_for_caleitc.py:1]: change back

File: policyengine_us/parameters/gov/states/ca/tax/income/credits/wftc/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/credits/wftc/amount.yaml:1]: change to reflect san Francisco in description and label

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/local/ca/sf/wftc/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  unit: currency-USD
+  period: year
+  label: San Francisco working families tax credit amount
+  reference:
+    - title: SF.Gov News 2024
+      href: https://www.sf.gov/news/san-francisco-launches-free-tax-preparation-assistance-and-local-tax-credit-program-and
+    - title: San Francisco Working Families Credits Application Tax Year 2022
+      href: https://www.sfhsa.org/sites/default/files/media/document/2023-01/form_wfc_ty22_eng_code41.pdf#page=4
+    - title: CHAPTER 12S-SEC. 12S.1. WORKING FAMILIES CREDIT PROGRAM
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/local/ca/sf/wftc/amount.yaml:13]:
```suggestion
    # The legal code states the San Francisco Human Services Agency shall publish the 
    # credit amount independently
    - title: CHAPTER 12S-SEC. 12S.1. WORKING FAMILIES CREDIT PROGRAM
```

File: policyengine_us/variables/gov/local/ca/sf/wftc/ca_sf_wftc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/local/ca/sf/wftc/ca_sf_wftc.py:1]: can we add this credit to the list of refundable credits in California

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ca/tax/income/credits/refundable.yaml [hunk 5]
```diff
@@ -3,14 +3,20 @@ values:
   2021-01-01:
     - ca_eitc
     - ca_yctc
+    - ca_sf_wftc
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/credits/refundable.yaml:6]: should this be in 2021? the parameters start in 2022

File: policyengine_us/variables/gov/local/ca/sf/wftc/ca_sf_wftc.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    entity = TaxUnit
+    label = "San Francisco Working Families Tax Credit"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.CA
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.local.ca.sf.wftc
+        eitc_eligible = tax_unit("eitc_eligible", period)
+        caeitc_eligible = tax_unit("ca_eitc_eligible", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/local/ca/sf/wftc/ca_sf_wftc.py:15]:
```suggestion
        ca_eitc_eligible = tax_unit("ca_eitc_eligible", period)
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ca/tax/income/credits/refundable.yaml [hunk 7]
```diff
@@ -7,10 +7,15 @@ values:
     - ca_foster_youth_tax_credit
     - ca_eitc
     - ca_yctc
+    - ca_sf_wftc
 metadata:
   reference:
     - title: California Form 540
       href: "https://www.ftb.ca.gov/forms/Search/Home/Confirmation"
+    # The legal code states the San Francisco Human Services Agency shall publish the 
+    # credit amount independently
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/credits/refundable.yaml:16]:
```suggestion
    # San Francisco working families tax credit
```

File: policyengine_us/variables/gov/local/ca/sf/wftc/ca_sf_wftc.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    entity = TaxUnit
+    label = "San Francisco Working Families Tax Credit"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.CA
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.local.ca.sf.wftc
+        county = tax_unit.household("county_str", period)
+        county_eligible = county == "SAN_FRANCISCO_COUNTY_CA"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/local/ca/sf/wftc/ca_sf_wftc.py:15]:
```suggestion
        in_sf = county == "SAN_FRANCISCO_COUNTY_CA"
```

**Review by PavelMakarchuk — APPROVED** [review]
good besides minor formatting comments

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
please add the 2023 tax year reference, both to the parameter and variable

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**